x-pack/platform/plugins/shared/lens/public/plugin.ts

```diff
@@ -386,12 +388,21 @@ export class LensPlugin {
         return createLensEmbeddableFactory(deps);
       });
 
-      embeddable.registerLegacyURLTransform(LENS_EMBEDDABLE_TYPE, async () => {
-        const { getLensTransforms } = await import('../common/transforms');
-        return getLensTransforms({
-          transformEnhancementsIn: embeddable.transformEnhancementsIn,
-          transformEnhancementsOut: embeddable.transformEnhancementsOut,
-        }).transformOut;
+      core.getStartServices().then(async ([{ featureFlags }]) => {
+        const flags = await setLensFeatureFlags(featureFlags);
+        void setLensBuilder();
+
+        embeddable.registerLegacyURLTransform(LENS_EMBEDDABLE_TYPE, async () => {
+          const { getLensTransforms } = await import('../common/transforms');
+          const { LensConfigBuilder } = await import('@kbn/lens-embeddable-utils');
+          const builder = new LensConfigBuilder(undefined, flags.apiFormat);
+
+          return getLensTransforms({
+            builder,
+            transformEnhancementsIn: embeddable.transformEnhancementsIn,
+            transformEnhancementsOut: embeddable.transformEnhancementsOut,
+          }).transformOut;
+        });
       });
```

Comment:
Just wondering: isn't better to configure this in the start method instead of the setup?
If I'm not wrong the setup is a sync method. or probably do the opposite: get the core start services within the registerLegacy URL async function?